modules/grpc/bigquery/bigquery-dest.cpp

```diff
@@ -77,19 +138,59 @@ DestinationDriver::format_persist_name()
 }
 
 const gchar *
-DestinationDriver::format_stats_instance()
+DestinationDriver::format_stats_key(StatsClusterKeyBuilder *kb)
 {
-  static gchar stats[1024];
+  stats_cluster_key_builder_add_legacy_label(kb, stats_cluster_label("driver", "bigquery"));
+  stats_cluster_key_builder_add_legacy_label(kb, stats_cluster_label("url", this->url.c_str()));
+  stats_cluster_key_builder_add_legacy_label(kb, stats_cluster_label("project", this->project.c_str()));
+  stats_cluster_key_builder_add_legacy_label(kb, stats_cluster_label("dataset", this->dataset.c_str()));
+  stats_cluster_key_builder_add_legacy_label(kb, stats_cluster_label("table", this->table.c_str()));
 
-  g_snprintf(stats, sizeof(stats), "bigquery,%s,%s,%s,%s", this->url.c_str(),
-             this->project.c_str(), this->dataset.c_str(), this->table.c_str());
+  return nullptr;
+}
 
-  return stats;
+void
+DestinationDriver::construct_schema_prototype()
+{
+  this->descriptor_pool.~DescriptorPool();
+  new (&this->descriptor_pool) google::protobuf::DescriptorPool();
```

Comment:
It's probably my lack of C++ knowledge, but can we not just do this?
```
this->descriptor_pool = google::protobuf::DescriptorPool();
```

Reply:
This is a dirty in-place constructor trick as the class does not have a copy constructor or an operator=.

Reply:
If it works it is okay for me :)

Consider this resolved.